docs/sources/k6/next/reference/versioning-and-stability-guarantees.md

```diff
@@ -0,0 +1,285 @@
+2. If necessary, add an alternative solution. Ideally, this would be an experimental version within a minor release within the current maintained state. Alternatively, it might be available in a Preview release.
+3. Provide documentation for the migration path.
+4. Proceed with removing the API and releasing a new major version.
+
+#### Update an API or a feature
+
+1. Apply the changes, then publish them in the next major version. If the change is significant, it might be convenient to make it available in a Preview release that can support the migration path until a version is globally available.
+2. Provide documentation for the migration path.
+
+### Deprecation
+
+When the k6 team intends to remove an API or a feature, it will be marked as *deprecated*. This occurs when an API becomes obsolete, superseded by another API, or otherwise discontinued for business or technological reasons.
+Deprecated APIs remain available through their deprecated phase, which guarantees they will remain available at least throughout the entire life cycle of the current major version.
+
+A deprecation will be announced through several layers of communication before it will be officially applied:
+
+1. Release notes and CHANGELOGs.
+2. Official documentation.
+3. When the user directly interacts with the feature.
+4. An alternative already available is recommended, but it shouldn’t be considered mandatory.
+
+### Migration
+
+When a breaking change is introduced, or an API is superseded, then documentation for a migration guide will be provided to users to help them understand how to move from version to version.
+The migration path will be available in the official documentation, and it will be mentioned in all other communication channels.
+
+## Support and development policies
+
+We define k6 support and development policies according to four states where a k6 version can be implemented. At any given time, we have:
+
+* **The under-development** version is the next upcoming version of k6. Note that this version might either start a new major version, for example, developing v3.0.0 while the latest release is v2.8.2, or it might be a new minor or patch iteration over the latest release.
+* **The actively maintained** version is the latest released major version of k6.
+* **The supported** version is the latest released version of k6 within the previous major version.
+* **The deprecated** versions are the set of previously released k6 versions.
+
+As such, during its development and support workflow, the k6 open source project actively develops the next version of k6, maintains the latest major release, supports the previous major version, and doesn't provide any support for releases older than the supported one.
+
+Any past version of k6 that’s neither being developed, maintained, or supported is considered **deprecated** and will not support further updates or development. Users are free to use them at their own risk.
```

Comment:
Should we explicitly state that bugfixes or security issues also apply there?

Reply:
I think it might be a good idea... Something along the lines of:

```suggestion
Any past version of k6 that’s neither being developed, maintained, or supported is considered **deprecated** and will not receive further updates or development, including bug fixes or security updates. Users are free to use them at their own risk.
```